### ISSUE TITLE
Unable to find windows.h

### ISSUE THREAD

**Issue by crlf0710** [nabijaczleweli/rust-embed-resource#11]
Well, on my machine it builds fine for `gnu` builds, however if i run cargo build under plain cmd.exe console, it can find "rc.exe" but can't find `windows.h`.  After "vcvarsall.bat" is run everything goes fine.

So i think maybe this library should try to detect and set up the environment in certain cases?

**Comment by crlf0710** [comment]
Well, that's fine. I'd be happy to fix it myself if it's not too much "dirty" hack work, however i'm not sure of a proper solution at this point. If @retep998 or someone else can mentor me or at least give me some hints it would be great.

**Comment by retep998** [comment]
I'd like to eventually get a crate going to be able to locate all these sort of VS things and then this crate could use that crate to find `rc.exe` and associated stuff.

**Comment by nabijaczleweli** [comment]
That'd be lovely, indeed!

**Comment by nabijaczleweli** [comment]
Released in [`v1.3.2`](https://github.com/nabijaczleweli/rust-embed-resource/releases/tag/v1.3.2), but do please update this thread with information on whether #17 fixed your builds!

**Comment by helgoboss** [comment]
@nabijaczleweli I had the same issue like the reporter. However, the PR didn't fix things for me, it still couldn't find `windows.h`. The `INCLUDE` env var was not set at all.

I figured that [cc](https://crates.io/crates/cc) crate must have some way of finding the correct env variables for building. Fortunately its API is exposed, so currently I use the following workaround which sets the `INCLUDE` env var correctly (and a bunch of other env vars probably not necessary for running `rc.exe`):

```rust
    if let Some(tool) = cc::windows_registry::find_tool(target.as_str(), "cl.exe") {
        for (key, value) in tool.env() {
            env::set_var(key, value);
        }
    }
    embed_resource::compile("foobar.rc");
```

**Comment by nabijaczleweli** [comment]
Right; not great, but still progress, I guess.

@helgoboss Would you be so kind so as to post the output of something to the effect of `println!("{} = {}", key, value);` as well as your VS version?

**Comment by helgoboss** [comment]
@nabijaczleweli It's VS 2017. Besides `INCLUDE` that sets `LIB` and `PATH` but turns out only `INCLUDE` is needed. This is its value:

```
C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\\VC\\Tools\\MSVC\\14.15.26726\\include;C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.17134.0\\ucrt;C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.17134.0\\um;C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.17134.0\\cppwinrt;C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.17134.0\\winrt;C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.17134.0\\shared;
```

**Comment by xobs** [comment]
> 
> 
> @nabijaczleweli I had the same issue like the reporter. However, the PR didn't fix things for me, it still couldn't find `windows.h`. The `INCLUDE` env var was not set at all.
> 
> I figured that [cc](https://crates.io/crates/cc) crate must have some way of finding the correct env variables for building. Fortunately its API is exposed, so currently I use the following workaround which sets the `INCLUDE` env var correctly (and a bunch of other env vars probably not necessary for running `rc.exe`):
> 
> ```rust
>     let target = std::env::var("TARGET").unwrap();
>     if let Some(tool) = cc::windows_registry::find_tool(target.as_str(), "cl.exe") {
>         for (key, value) in tool.env() {
>             std::env::set_var(key, value);
>         }
>     }
>     embed_resource::compile("foobar.rc");
> ```

I can confirm that this fixes the build for me, and I'll include this as a workaround.

I'm using Visual Studio Community 2019, and I'm trying to compile `libui` without using `cmake`.  Without the fix where I modify env, I get this error:

```
rror: failed to run custom build command for `ui-sys v0.2.1 (D:\Code\libui-rs\ui-sys)`

Caused by:
  process didn't exit successfully: `D:\Code\libui-rs\target\debug\build\ui-sys-d8024265c3b43cdf\build-script-build` (ex
it code: 101)
--- stdout
Microsoft (R) Windows (R) Resource Compiler Version 6.3.9600.17336
Copyright (C) Microsoft Corporation.  All rights reserved.

libui/windows/winapi.hpp(34) : fatal error RC1015: cannot open include file 'windows.h'.

--- stderr
error: 'rustfmt.exe' is not installed for the toolchain 'nightly-x86_64-pc-windows-msvc'
To install, run `rustup component add rustfmt --toolchain nightly-x86_64-pc-windows-msvc`
error: backtrace:
error: stack backtrace:
   0: <unknown>
   1: <unknown>
   2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: <unknown>
   7: <unknown>
   8: BaseThreadInitThunk
   9: RtlUserThreadStart
```

The workaround results in a working UI.

**Comment by roblabla** [comment]
I'm hitting the same issue, trying to include `<winuser.h>` to avoid having to define RT_MANIFEST myself (and to copy paste example code from microsoft).

So, given embed-resource depends on cc for `llvm-rc`, I suppose it could do the heavy-lifting of figuring out what the includes are for us, E.G. replacing [this function](https://github.com/nabijaczleweli/rust-embed-resource/blob/master/src/windows_msvc.rs#L143) with a call to `cc::find_tool().env()`, as suggested [here](https://github.com/nabijaczleweli/rust-embed-resource/issues/11#issuecomment-609336378). Then `embed-resource` would only be responsible for the peculiarity of locating `rc.exe` and running it with the right arguments.

Would this be an acceptable solution?

**Comment by nabijaczleweli** [comment]
Fix released in [`v1.6.1`](https://github.com/nabijaczleweli/rust-embed-resource/releases/tag/v1.6.1).
